src/jobs/candlepinPipelineJob.groovy

```diff
@@ -1,49 +1,54 @@
 import jobLib.rhsmLib
 
 String baseFolder = rhsmLib.candlepinJobFolder
-String githubOrg = binding.variables['CANDLEPIN_JENKINS_GITHUB_ORG'] ?: 'candlepin'
 
-pipeline_helper = job("$baseFolder/candlepin-pipeline-helper") {
-    environmentVariables {
-        groovy('''
-            return [emailDestination: binding.variables.get('ghprbPullAuthorEmail') ?: binding.variables.get('GIT_AUTHOR_EMAIL')]
-        ''')
-    }
-    logRotator {
-        numToKeep(1)
-    }
-    steps {
-        downstreamParameterized {
-            trigger('candlepin') {
-                parameters {
-                    predefinedProp('sha1', '${sha1}')
-                    predefinedProp('ghprbActualCommit', '${ghprbActualCommit}')
-                    predefinedProp('emailDestination', '${emailDestination}')
-                    predefinedProp('ghprbTargetBranch', '${ghprbTargetBranch}')
-                    predefinedProp('ghprbPullId', '${ghprbPullId}')
+multibranchPipelineJob("${baseFolder}/candlepin") {
+    branchSources {
+        // upstream; PRs only
+        branchSource {
+            source {
+                github {
+                    id('9dff2e67-8a82-415b-afc7-f79fe76795cd') // IMPORTANT: use a constant and unique identifier
+                    credentialsId('github-api-token-as-username-password')
+                    repoOwner('candlepin')
+                    repository('candlepin')
+                    repositoryUrl('https://github.com/candlepin/candlepin.git')
+                    configuredByUrl(false)
+                    traits {
+                        gitHubPullRequestDiscovery {
+                            strategyId(1)
+                        }
+                        cleanAfterCheckoutTrait {
+                            extension {
+                                deleteUntrackedNestedRepositories(true)
+                            }
+                        }
+                    }
                 }
             }
         }
     }
-}
-
-pipeline = pipelineJob("$baseFolder/candlepin") {
-    description('Delivery Pipeline for candlepin')
-    parameters {
-        stringParam('sha1', 'master', 'GIT commit hash of what you want to test.')
-        stringParam('ghprbActualCommit', null, 'commit used to report status against a GitHub PR')
-        stringParam('ghprbTargetBranch', null, 'PR target branch')
-        stringParam('emailDestination', null, 'email address to report failures to')
-        stringParam('ghprbPullId', null, 'PR number (used in certain checks)')
+    factory {
+        workflowBranchProjectFactory {
+            scriptPath('jenkins/Jenkinsfile')
+        }
     }
-    logRotator {
-        numToKeep(20)
+    orphanedItemStrategy {
+        discardOldItems {
+            numToKeep(5)
+        }
     }
-    definition {
-        cps {
-            script(readFileFromWorkspace('src/resources/candlepinPipeline.groovy'))
+    triggers {
+        periodicFolderTrigger {
+            interval('5m')
```

Comment:
So this scans every 5 minutes for new PRs and launches a new pipeline for each one?

Reply:
@nikosmoum  Correct.  We can change this to whatever time interval you want, but I think 5 minutes is a good balance of rapid turn around and not constantly using resources just to scan.

Also we can scan for specific branch names (or a regex) and test upon changes to those if you want that.